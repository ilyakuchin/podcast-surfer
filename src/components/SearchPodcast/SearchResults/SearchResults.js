import React, { Component } from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";

const Podcasts = styled.ul`
  margin: 0;
  padding: 0;
  margin-top: 50px;
  grid-area: results;
  list-style-type: none;
`;

const PodcastGrid = styled.li`
  margin-top: 50px;
  justify-items: center;
  align-items: center;
  max-width: 700px;
  display: grid;
  grid-template-areas: "podcast-image podcast-link";
  justify-items: center;
  align-items: center;
  grid-template-columns: 200px auto;

  @media (max-width: 600px) {
    grid-template-areas:
      "podcast-image"
      "podcast-link";
  }
`;

const PodcastImage = styled.img`
  grid-area: podcast-image;
  width: 200px;
  height: 200px;
  object-fit: scale-down;
`;

const PodcastLink = styled.div`
  grid-area: podcast-link;

  @media (max-width: 600px) {
    text-align: center;
  }
`;

export default class SearchResults extends Component {
  render() {
    return (
      <Podcasts>
        {this.props.podcasts.map(item => (
          <PodcastGrid key={item.id}>
            <PodcastImage src={item.image} alt="podcast cover" />
            <PodcastLink>
              <Link to={{ pathname: "/podcast", state: { rss: item.rss } }}>
                {item.name}
              </Link>
            </PodcastLink>
          </PodcastGrid>
        ))}
      </Podcasts>
    );
  }
}
