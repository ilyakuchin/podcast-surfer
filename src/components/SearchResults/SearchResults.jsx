import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Message, Grid } from 'semantic-ui-react';
import ConnectedPodcastList from '../PodcastList/PodcastList';

export function SearchResults({ podcasts, isFetching, error }) {
  return (
    <div>
      {error !== '' ? (
        <Grid textAlign='center' columns={1} container stackable>
          <Grid.Column style={{ maxWidth: 450 }}>
            <Message negative>
              <Message.Header>Error</Message.Header>
              <p>{error}</p>
            </Message>
          </Grid.Column>
        </Grid>
      ) : (
        <ConnectedPodcastList podcasts={podcasts} isFetching={isFetching} />
      )}
    </div>
  );
}

const mapStateToProps = state => {
  return {
    podcasts: state.podcasts.podcasts,
    isFetching: state.podcasts.isFetching,
    error: state.podcasts.error
  };
};

SearchResults.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  podcasts: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      image: PropTypes.string,
      rss: PropTypes.string,
      name: PropTypes.string
    })
  ).isRequired,
  error: PropTypes.string.isRequired
};

const ConnectedSearchResults = connect(mapStateToProps)(SearchResults);

export default ConnectedSearchResults;
