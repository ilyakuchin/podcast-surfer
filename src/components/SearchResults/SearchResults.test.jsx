import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SearchResults } from './SearchResults';

configure({ adapter: new Adapter() });

describe('Test SearchResults Component', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<SearchResults isFetching podcasts={[]} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
