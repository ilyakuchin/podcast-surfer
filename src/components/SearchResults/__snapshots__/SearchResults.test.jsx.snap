// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchResults Component should render correctly 1`] = `
<div>
  <Connect(PodcastList)
    isFetching={true}
    podcasts={Array []}
  />
</div>
`;

exports[`Test SearchResults Component should render correctly 2`] = `
<div>
  <Connect(PodcastList)
    isFetching={false}
    podcasts={Array []}
  />
</div>
`;
